### ISSUE TITLE
approach to handle webhooks

### ISSUE THREAD

**Issue by fluxsaas** [ekomi-ltd/fastbill-automatic#6]
first approach, what do you think?

Webhooks:
https://github.com/easyPEP/fastbill-automatic/commit/d6414fe5a6491f506cf63f45f47903b2dcb3fec9

Payments:
https://github.com/easyPEP/fastbill-automatic/commit/558ce187eb7e15dbb609bcf99e1de5d4db960ae9

i need this functionality to trigger some action on the accounts, like:
- show hint: invoice reminder
- show hint: update payment information
- account goes to a lockdown state (invoice not paid / failed)
- trial ended
- account information updated
- invoice paid
- ...

**Comment by skorth** [comment]
Hey @fluxsaas. Without testing it, what happens if you use Fastbill::Automatic::Payment.get(). Does the API support this?

**Comment by fluxsaas** [comment]
nope. but i guess in the future. i added it because of the "completeness" for the webhooks (the body has a payment part in it. See https://github.com/easyPEP/fastbill-automatic/commit/d6414fe5a6491f506cf63f45f47903b2dcb3fec9#diff-e6d6fedb4200419e516456ce1d19e44dR70). 

i'm in contact with the developers and i'm sure they would provide us with the functionality if neceserry. 

IMHO i can't see any case where it should be relevant...

**Comment by fluxsaas** [comment]
well, we do not need to inherit from Base...

**Comment by fluxsaas** [comment]
example for webhooks with response:

https://gist.github.com/fluxsaas/6974639

^ i got those from fasbill

**Comment by skorth** [comment]
I am not sure if parsing the webhook response should be done within the gem (not sure at the moment). If fastbill would allow to configure also webhooks i would see a use case. For example Webhook.new(url: 'https://myapi....', event_types: ['eventtype1', 'eventtype2']). But still, its just something to map specific events on specific urls. Let me think about it.

**Comment by fluxsaas** [comment]
alright.

**Comment by skorth** [comment]
Just wait some days, maybe its a good idea. Just need some days to think about it.

**Comment by skorth** [comment]
Hey @fluxsaas, is there a reason why you use (just one example) `instance_variable_set("@#{key.downcase}", Customer.new(value))` instead of just  `instance_variable_set("@customer", Customer.new(value))`?
Would be nice if you change the Payment class, that it doesn't inherit from base or add a missing method for get(). I asked Fastbill and they didn't plan to add an API call for Payments. I would merge it if you added these small changes.

**Comment by fluxsaas** [comment]
alright, i updated the code. i gonna add more webhooks when i implement the functionality in my app.

**Comment by skorth** [comment]
Cool, thx.
